pkg/pod/entrypoint_lookup_impl_test.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pod
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/registry"
+	remotetest "github.com/tektoncd/pipeline/test"
+	"github.com/tektoncd/pipeline/test/parse"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	fakeclient "k8s.io/client-go/kubernetes/fake"
+)
+
+const (
+	username = "foo"
+	password = "bar"
+)
+
+type fakeHTTP struct {
+	reg http.Handler
+}
+
+func newfakeHTTP() fakeHTTP {
+	reg := registry.New()
+	return fakeHTTP{
+		reg: reg,
+	}
+}
+
+func (f *fakeHTTP) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// Request authentication for ping request
+	// For further reference see https://docs.docker.com/registry/spec/api/#api-version-check
+	if r.URL.Path == "/v2/" && r.Method == http.MethodGet {
+		w.Header().Add("WWW-Authenticate", "basic")
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+	// Check auth if we've fetching the image
+	if strings.HasPrefix(r.URL.Path, "/v2/task") && r.Method == "GET" {
+		u, p, ok := r.BasicAuth()
+		if !ok || username != u || password != p {
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+	}
+	// Default to open
+	f.reg.ServeHTTP(w, r)
+}
+
+func generateSecret(host string, username string, password string) *corev1.Secret {
+	return &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "secret",
+			Namespace: "ns",
+		},
+		Type: corev1.SecretTypeDockercfg,
+		Data: map[string][]byte{
+			corev1.DockerConfigKey: []byte(
+				fmt.Sprintf(`{%q: {"auth": %q}}`,
+					host,
+					base64.StdEncoding.EncodeToString([]byte(username+":"+password)),
+				),
+			),
+		},
+	}
+}
+
+func TestGetImageWithImagePullSecrets(t *testing.T) {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	ftp := newfakeHTTP()
+	s := httptest.NewServer(&ftp)
+	defer s.Close()
+
+	u, err := url.Parse(s.URL)
+	if err != nil {
+		t.Errorf("Parsing url with an error: %v", err)
+	}
+
+	task := parse.MustParseTask(t, `metadata:name: test-create-image`)
```

Comment:
```suggestion
	task := &v1beta1.Task{Name: "test-create-image"}
```

Reply:
This code suggestion will cause error, but I find something similar.

Reply:
Most of the time I just free hand these suggestions without running it through a compiler, so no worries if this isn't exact! 😅 

Key thing here is there's no need to parse yaml into a struct object - we can just define the struct object. This can be useful for in other tests to cut down on some of the verbosity, or to reference other example files, but for trivial cases like this there isn't that much benefit.